### PR TITLE
Added support for input=text

### PR DESCRIPTION
**Pull request by vsavkin**

### CODE REVIEW

File: modules/angular2/src/forms/directives.js [hunk 1]
```diff
@@ -1,58 +1,80 @@
...
 
-@CONST()
-class CheckboxControlValueAccessor extends ControlValueAccessor {
-  constructor() {
-    super();
+@Decorator({
+  selector: 'input[type=text]', //should be input[type=text][control]
+  // change the selector once https://github.com/angular/angular/issues/1025 is fixed
+  events: {
+    'change' : 'onChange($event.target.value)',
```
Comment (ajoslin) [inline modules/angular2/src/forms/directives.js]: Shouldn't we use the `input` event? As far as I understand, it's more reliable than change, keyup, or any combination of those. It detects everything.

Comment (vsavkin) [inline modules/angular2/src/forms/directives.js]: Fixed

File: modules/angular2/test/forms/integration_spec.js [hunk 2]
```diff
@@ -107,69 +112,67 @@ export function main() {
...
+          expect(input.value).toEqual("two");
+          async.done();
+        });
+      }));
 
-    if (!IS_NODEJS) {
       describe("different control types", () => {
-        it("should support type=checkbox", inject([AsyncTestCompleter], (async) => {
-          var ctx = new MyComp(new ControlGroup({"checkbox": new Control(true)}));
+        it("should support type=text", inject([AsyncTestCompleter], (async) => {
```
Comment (vicb) [inline modules/angular2/test/forms/integration_spec.js]: @vsavkin I think we should also support `<input>` without explicit type (the default being "text"). I haven't look at the details to know if this is supported ? just leaving a note as a potential reminder here.

Comment (vsavkin) [inline modules/angular2/test/forms/integration_spec.js]: Good point. Currently our compiler's css selector engine is limited, which makes expressing such things hard. I will add a note.

Comment (vsavkin) [inline modules/angular2/test/forms/integration_spec.js]: I actually fixed  and added a test checking that it works

File: modules/angular2/test/forms/integration_spec.js [hunk 3]
```diff
@@ -107,74 +112,91 @@ export function main() {
...
+            input.value = "new";
+            dispatchEvent(input, "input");
+
+            expect(ctx.form.value).toEqual({"text": "new"});
+            async.done();
+          });
+        }));
 
-            expect(ctx.form.value).toEqual({"checkbox" : false});
+        it("should support <input> without type", inject([AsyncTestCompleter], (async) => {
```
Comment (vicb) [inline modules/angular2/test/forms/integration_spec.js]: :+1:

**Comment by vsavkin** [comment]
Hey @rkirov. Could you review this PR?

**Pull request merged**